File: src_original/extras-knewstuff/liberrorcode.cpp
#include <KNSCore/ErrorCode>
#include <QMetaObject>
#include <errorcode.h>
#include "liberrorcode.h"
#include "liberrorcode.hxx"

QMetaObject* KNSCore__ErrorCode_GetEnumMetaObject(int param1) {
    return (QMetaObject*)KNSCore::ErrorCode::qt_getEnumMetaObject(static_cast<KNSCore::ErrorCode::ErrorCode>(param1));
}

const char* KNSCore__ErrorCode_GetEnumName(int param1) {
    return (const char*)KNSCore::ErrorCode::qt_getEnumName(static_cast<KNSCore::ErrorCode::ErrorCode>(param1));
}

File: include_original/extras-knewstuff/liberrorcode.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const errorcode_enums = enums;
const std = @import("std");

/// https://api.kde.org/knscore-errorcode.html
pub const knscore__errorcode = struct {
    /// [Qt documentation](https://api.kde.org/knscore-errorcode.html#qt_getEnumMetaObject)
    ///
    /// ``` param1: errorcode_enums.ErrorCode ```
    pub fn GetEnumMetaObject(param1: i32) QtC.QMetaObject {
        return qtc.KNSCore__ErrorCode_GetEnumMetaObject(@intCast(param1));
    }

    /// [Qt documentation](https://api.kde.org/knscore-errorcode.html#qt_getEnumName)
    ///
    /// ``` param1: errorcode_enums.ErrorCode ```
    pub fn GetEnumName(param1: i32) []const u8 {
        const _ret = qtc.KNSCore__ErrorCode_GetEnumName(@intCast(param1));
        return std.mem.span(_ret);
    }
};

/// https://api.kde.org/knscore-errorcode.html#types
pub const enums = struct {
    pub const ErrorCode = enum {
        pub const UnknownError: i32 = 0;
        pub const NetworkError: i32 = 1;
        pub const OcsError: i32 = 2;
        pub const ConfigFileError: i32 = 3;
        pub const ProviderError: i32 = 4;
        pub const InstallationError: i32 = 5;
        pub const ImageError: i32 = 6;
        pub const AdoptionError: i32 = 7;
        pub const TryAgainLaterError: i32 = 8;
    };
};
